pkg/generate/generator/sourceref.go

```diff
@@ -48,7 +48,7 @@ func (g *SourceRefGenerator) FromDirectory(directory string) (*app.SourceRef, er
 	// Make sure that this is a git directory
 	gitRoot, err := g.repository.GetRootDir(directory)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not detect git repository in %s", directory)
```

Comment:
Is the only reason GetRootDir can fail due to the directory not being there?  What if it's permission denied, or a transient filesystem error?